src/BuildConfig.cc

```diff
@@ -97,6 +97,99 @@ BuildConfig::BuildConfig(const std::string& packageName, const bool isDebug)
   }
 }
 
+// Generally split the string by space character, but it will properly interpret
+// the quotes and some escape sequences. (More specifically it will ignore
+// whatever character that goes after a backslash and preserve all characters,
+// usually used to pass an argument containing spaces, between quotes.)
+static std::vector<std::string>
+parseEnvFlags(std::string_view env) {
+  std::vector<std::string> result;
+
+  // This flag will be set when a backslash encountered.
+  bool ignoreNext = false;
+
+  // This flag will be set when quote encountered.
+  // The quote encountered will be stored in the ignoreNextSeqStart to
+  // distinguish different quote types.
+  bool ignoreNextSeq = false;
+  char ignoreNextSeqStart = ' ';
+
+  result.emplace_back();
+
+  // There are bunches of conditional predicates, and to avoid if-else chain the
+  // special cases will be processed first and end in `continue` to skip rest of
+  // the loop. Same in special cases that are in another special case.
+  for (char i : env) {
+
+    if (ignoreNext) {
+      result.back() += i;
```

Comment:
Let's use `std::string buffer`, and then this can be clearer. We also don't need to do `result.emplace_back()`.